src/IROperator.cpp

```diff
@@ -536,13 +568,8 @@ void match_types_bitwise(Expr &x, Expr &y, const char *op_name) {
         internal_assert(x.type().lanes() == y.type().lanes()) << "Can't match types of differing widths";
     }
 
-    // Cast to the wider type of the two. Already guaranteed to leave
-    // signed/unsigned on number of lanes unchanged.
-    if (x.type().bits() < y.type().bits()) {
-        x = cast(y.type(), x);
-    } else if (y.type().bits() < x.type().bits()) {
-        y = cast(x.type(), y);
-    }
+    // Cast to the wider type of the two.
+    match_bits(x, y);
```

Comment:
Do we want this change in all cases `match_types_bitwise` is used, or just shifts? This change will affect bitwise and, or, and xor too.

Reply:
I had been trying to reduce repeated code, but it's probably simpler to let this stay as it's own case.